$MirOS: ports/devel/glib/patches/patch-glib_gtimer_c,v 1.1 2008/03/06 22:08:24 tg Exp $
--- glib/gtimer.c.orig	Mon Jul 16 20:56:49 2007
+++ glib/gtimer.c	Mon Mar  2 18:56:36 2009
@@ -390,6 +390,7 @@ gchar *
 g_time_val_to_iso8601 (GTimeVal *time_)
 {
   gchar *retval;
+  time_t real_tv_sec;
 
   g_return_val_if_fail (time_->tv_usec >= 0 && time_->tv_usec < G_USEC_PER_SEC, NULL);
 
@@ -397,9 +398,11 @@ g_time_val_to_iso8601 (GTimeVal *time_)
 #define ISO_8601_FORMAT "%Y-%m-%dT%H:%M:%SZ"
   retval = g_new0 (gchar, ISO_8601_LEN + 1);
   
+  /* time_->tv_sec is of type glong, not time_t */
+  real_tv_sec = time_->tv_sec;
   strftime (retval, ISO_8601_LEN,
 	    ISO_8601_FORMAT,
-	    gmtime (&(time_->tv_sec)));
+	    gmtime (&real_tv_sec));
   
   return retval;
 }
